projects/libs/core-library-typescript/src/transaction-pool/transaction-validator.ts

```diff
@@ -31,19 +44,27 @@ export class TransactionValidator {
     const status = await this.subscriptionManager.getSubscriptionStatus({ subscriptionKey });
 
     if (!status.active) {
-      logger.info("subscription ${subscriptionKey} not active");
+      logger.info(`Subscription ${subscriptionKey} not active`);
     }
 
     return status.active;
   }
 
+  private verifySignature(tx: types.Transaction) {
+    const txHelper = new TransactionHelper(tx);
+    const hash = txHelper.calculateHash();
+    const key = ec.keyFromPublic(tx.signatureData.publicKey.toString("hex"));
+    return key.verify(hash, tx.signatureData.signature.toString("hex"));
```

Comment:
Very small suggestion: `verify` can throw, which might crash the service, so perhaps it'd be better to wrap it via `try/catch`.